File: javascript/grade-school/grade-school.js
export class GradeSchool {
    constructor() {
        this.DB = {};
    }

    roster() {
        return Object.keys(this.DB)
            .reduce((copy, grade) => {
                copy[grade] = this.grade(grade);
                return copy;
            }, {});
    }

    add(name, grade) {
        this.DB[grade] = this.grade(grade).concat(name).sort();
    }

    grade(grade) {
        return Object.assign([], this.DB[grade]);
    }
}

File: python/grade-school/grade_school.py
from collections import defaultdict

class School:
    def __init__(self):
        self.database = defaultdict(set)

    def add_student(self, name, grade):
        self.database[grade].add(name)

    def roster(self):
        return [name
                for grade in sorted(self.database)
                for name in self.grade(grade)]

    def grade(self, grade):
        return sorted(self.database[grade])
